<a href="https://colab.research.google.com/github/Vronska-Anhelina/Statistical-Methods-for-Business/blob/main/Website_Traffic_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np


visitors=[150,200, 180, 220,	190,	250,	170, 1600, 210,	230]
range=max(visitors)-min(visitors)
print(f"Range of visitors:{range}")

Range of visitors:1450


In [2]:
visitor_sort=sorted(visitors)
visit_left=visitor_sort[: len(visitor_sort)//2]
visit_right=visitor_sort[len(visitor_sort)//2:]
q1=np.median(visit_left)
q2=np.median(visitor_sort)
q3=np.median(visit_right)

print(f"Q1:{q1}")
print(f"Median(Q2):{q2}")
print(f"Q3:{q3}")

Q1:180.0
Median(Q2):205.0
Q3:230.0


In [3]:
IQR=q3-q1
print(IQR)

50.0


In [4]:
std_dev = np.std(visitor_sort)
print(f"standard deviation:{std_dev}")


standard deviation:420.9275472097306


Using the calculations above, we can say that the data is quite variable. This is indicated by the large value of the range. It also indicates instability in site visits during a given period. The larger the range indicator, the greater the instability that can be seen in visits. The interquartile range is 50, which indicates a narrow range. Comparing it with the range of the entire data set and the interquartile range, we can say that there are anomalous values ​​in the set (in our case 1600), which affect the calculation of the range (the range is sensitive to outliers, unlike the interquartile range). The typical value of the number of visitors is the median of this sample. That is, 205 visitors. The standard deviation shows that most of the values ​​​​variate by 420 units from the mean value. This indicates extreme values ​​​​(outliers) in the set and the sensitivity of the standard deviation to such values.